lib/gpu/gpuProcess.cpp

```diff
@@ -152,23 +155,27 @@ void gpuProcess::main_thread() {
     bool first_run = true;
 
     while (!stop_thread) {
-        int gpu_frame_id = gpu_frame_counter % _gpu_buffer_depth;
+        int icommand = gpu_frame_counter % _gpu_buffer_depth;
```

Comment:
I'm not sure this works in the HSA case?   Since there we would only be creating one command per buffer.

Reply:
oh yes, you're right, of course.

Hmm, let's see, I could mod by the actual number of command objects in each vector, since currently it's up to the gpuProcess subclass to create that array of commands in `create_command`, and currently `main` is just assuming that it created `_gpu_buffer_depth` of them...

Reply:
Or hsaProcess could just make `_gpu_buffer_depth` copies of each command ;)

Reply:
okay, pushed a commit for that, I think it's better